include/exadg/structure/time_integration/time_int_gen_alpha.cpp

```diff
@@ -126,16 +126,18 @@ TimeIntGenAlpha<dim, Number>::do_timestep_solve()
   rhs.reinit(displacement_n);
   this->compute_const_vector(rhs, displacement_n, velocity_n, acceleration_n);
   pde_operator->apply_mass_operator(const_vector, rhs);
-  // set entries of constant vector corresponding to Dirichlet degrees of freedom
-  // to zero in order to allow convergence of solvers (especially for Newton solver,
-  // linear solver should converge without this line because the linear operator
-  // has values of 1 on the diagonal for constrained degrees of freedom).
-  pde_operator->set_constrained_values_to_zero(const_vector);
 
   if(param.large_deformation == false) // linear case
   {
     // calculate right-hand side vector
     pde_operator->compute_rhs_linear(rhs, this->get_mid_time());
+
+    // Constrained degrees of freedom have already been set
+    // correctly in the function compute_rhs_vector() for the rhs vector.
+    // Hence, the following line ensures that adding const_vector to rhs
+    // does not alter the constrained degrees of freedom of rhs.
+    pde_operator->set_constrained_values_to_zero(const_vector);
+
     // shift const_vector to right-hand side
     rhs.add(-1.0, const_vector);
```

Comment:
Linear problems: This should be refactored/redesigned in a follow-up PR. I think we can improve the logic for imposing inhomogeneous Dirichlet BCs also in case of linear problems.